### PR TITLE
Add JMX counter for cpu time in shard compaction

### PR DESCRIPTION
**Pull request by nileema**

### CODE REVIEW

File: presto-raptor/pom.xml [hunk 1]
```diff
@@ -191,7 +192,6 @@
         <dependency>
             <groupId>com.facebook.presto</groupId>
             <artifactId>presto-main</artifactId>
```
Comment (cberner) [inline presto-raptor/pom.xml]: Move this out of the test section. Is @electrum ok with making Raptor an internal plugin?

File: presto-raptor/src/main/java/com/facebook/presto/raptor/storage/ShardCompactionManager.java [hunk 2]
```diff
@@ -23,6 +23,7 @@
 import com.facebook.presto.raptor.metadata.TableMetadata;
 import com.facebook.presto.spi.NodeManager;
 import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.util.CpuTimer;
```
Comment (cberner) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/storage/ShardCompactionManager.java]: In my opinion it's not worth making Raptor an internal plugin just to reuse this. I would just copy it out of presto-main

**Comment by cberner** [comment]
Looks good, but see comments

**Comment by electrum** [comment]
Capitalize "CPU" in commit message

File: presto-raptor/pom.xml [hunk 3]
```diff
@@ -65,6 +65,7 @@
         <dependency>
             <groupId>io.airlift</groupId>
             <artifactId>units</artifactId>
+            <scope>provided</scope>
```
Comment (electrum) [inline presto-raptor/pom.xml]: This should NOT be provided as it is not part of the SPI.

File: presto-raptor/src/main/java/com/facebook/presto/raptor/util/CpuTimer.java [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+import io.airlift.units.Duration;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadMXBean;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+// This is a copy of com.facebook.presto.util.CpuTimer to avoid a dependency on presto-main
+public class CpuTimer
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/util/CpuTimer.java]: Remove all the unused methods?

Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/util/CpuTimer.java]: Let's add this to Airlift stats. Mark the API as `@Beta`

**Comment by electrum** [comment]
Looks good

**Pull request merged**